### PR TITLE
Replace non-standard topics

### PR DESCRIPTION
**Pull request by stkent**
Per #297 - normalizes topics from that issue (`sublist` topics have apparently been normalized since that issue was opened).

### CODE REVIEW

**Comment by stkent** [comment]
Should pass after #323 is merged.

**Comment by dector** [comment]
@stkent Easy fix for: `sublist` (`enumerations`) can be added as well. :)

File: config.json [hunk 1]
```diff
@@ -823,8 +823,8 @@
       "difficulty": 10,
       "topics": [
         "generics",
-        "higher_order_functions",
-        "nested_classes",
+        "functional_programming",
+        "classes",
```
Comment (dector) [inline config.json]: @eparovyshnaya what do you think about this. Should we use standard topics here?
I'm kinda ok with it.

Comment (eparovyshnaya) [reply config.json]: That's all about fitting to common metadata - it's mandatory. The other question is if this metadata rich enough to express significant pecularities. 

Both these cases are ok for me. `higher_order_functions` practically equals to `functional_programming` support, and `classes` covers `nested classes` with an appropriate context.

**Comment by dector** [comment]
Triggering Travis re-check.

**Pull request merged**